pkg/sdk/go/errors/errors.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package errors
```

Comment:
Since this is practically a rewriting of the MF errors package (because the implementation details from our errors are not exported), we should probably move this implementation to `pkg/errors/sdk_errors.go` and do something like:

```go
package errors

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
)

var (
	errFailedToReadBody = New("failed to read http response body ")
	errRespBodyNotJSON  = New("response body is not vaild JSON ")
	errJSONKeyNotFound  = New("response body expected error message json key not found ")
)

// SDKError type for SDK
type SDKError struct {
	*customError
	statusCode int
	status     string
}

func (e *SDKError) StatusCode() int {
	return e.statusCode
}

func (e *SDKError) Status() string {
	return e.status
}

// NewSDKErr returns the SDK error that formats as the given text.
func NewSDKErr(text string) SDKError {
	return SDKError{
		customError: &customError{
			msg: text,
			err: nil,
		},
	}
}

// NewWithStatus returns an Error that formats as the given text.
func NewWithStatus(statusCode int, status, text string) SDKError {
	return SDKError{
		statusCode: statusCode,
		status:     status,
		customError: &customError{
			msg: text,
			err: nil,
		},
	}
}

func EncodeError(resp *http.Response, errorMsgJsonKey string, expectedStatusCodes ...int) error {
	for expectedStatusCode := range expectedStatusCodes {
		if resp.StatusCode == expectedStatusCode {
			return nil
		}
	}

	b, bErr := io.ReadAll(resp.Body)
	if bErr != nil {
		e := Wrap(errFailedToReadBody, bErr)
		return Wrap(NewWithStatus(resp.StatusCode, resp.Status, ""), e)
	}

	if errorMsgJsonKey == "" {
		return NewWithStatus(resp.StatusCode, resp.Status, string(b))
	}

	var content map[string]interface{}
	err := json.Unmarshal(b, &content)
	if err != nil {
		e := Wrap(errRespBodyNotJSON, New(string(b)))

		return Wrap(NewWithStatus(resp.StatusCode, resp.Status, ""), e)
	}

	if msg, ok := content[errorMsgJsonKey]; ok {
		return NewWithStatus(resp.StatusCode, resp.Status, fmt.Sprintf("%v", msg))
	}
	e := Wrap(errJSONKeyNotFound, New(string(b)))
	return Wrap(NewWithStatus(resp.StatusCode, resp.Status, ""), e)
}
```
It's OK to have this in `pkg/errors` because that package does not have any other complex imports to pollute our SDK (and we already have it as a dependency).

Reply:
moved to pkg/errors